samples/csharp/getting-started/Ranking_PersonalizedSort/PersonalizedRanking/Program.cs

```diff
@@ -0,0 +1,190 @@
+﻿using Microsoft.ML;
+using Microsoft.ML.Trainers.LightGbm;
+using PersonalizedRanking.Common;
+using PersonalizedRanking.DataStructures;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Net;
+using static Microsoft.ML.DataOperationsCatalog;
+
+namespace PersonalizedRanking
+{
+    class Program
+    {
+        const string AssetsPath = @"../../../Assets";
+        readonly static string TrainDatasetPath = Path.Combine(AssetsPath, "InputData_Train.csv");
+        readonly static string TestDatasetPath = Path.Combine(AssetsPath, "InputData_Test.csv");
+        readonly static string ModelPath = Path.Combine(AssetsPath, "RankingModel.csv");
+
+        readonly static string OriginalDatasetPath = Path.Combine(AssetsPath, "Train.csv");
+        readonly static string OriginalExampleDatasetPath = Path.Combine(AssetsPath, "Test.csv");
+
+        static void Main(string[] args)
+        {
+            // Create a common ML.NET context.
+            // Seed set to any number so you have a deterministic environment for repeateable results.
+            MLContext mlContext = new MLContext(seed: 0);
+
+            try
+            {
+                PrepDatasets(mlContext, AssetsPath, OriginalDatasetPath, TrainDatasetPath, TestDatasetPath);
+
+                var model = TrainModel(mlContext, TrainDatasetPath, ModelPath);
+
+                EvaluateModel(mlContext, model, TestDatasetPath);
+
+                ConsumeModel(mlContext, ModelPath, OriginalExampleDatasetPath);
+            }
+            catch (Exception e)
+            {
+                Console.WriteLine(e.Message);
+            }
+
+            Console.ReadLine();
+        }
+
+        static void PrepDatasets(MLContext mlContext, string assetPath, string originalDatasetPath, string trainDatasetPath, string testDatasetPath)
+        {
+            const string DatasetUrl = "https://www.kaggle.com/c/expedia-personalized-sort/download/data.zip";
+
+                if (!File.Exists(trainDatasetPath) || !File.Exists(testDatasetPath))
+                {
+                    if (!File.Exists(originalDatasetPath))
+                    {
+                        throw new InvalidOperationException($"This samples requires the Expedia dataset.  Please ensure that you have downloaded and extracted the contents of the .zip file to the following directory: {assetPath}. The .zip file can be downloaded from here: {DatasetUrl}");
+                    }
+
+                    Console.WriteLine("===== Prepare the testing/training datasets =====");
+
+                    // Load dataset using TextLoader by specifying the type name that holds the data's schema to be mapped with datasets.
+                    IDataView data = mlContext.Data.LoadFromTextFile<HotelData>(originalDatasetPath, separatorChar: ',', hasHeader: true);
```

Comment:
Where do `HotelData` and `HotelPrediction` come from? I don't see them in this PR.

Reply:
They are in the DataStructures directory - I see them in the PR...maybe you overlooked it?  Anyway, since moving to the new dataset, these files have been removed and replaced with new types.

Reply:
#resolve